File: backend/app/schemas/content.py
from pydantic import BaseModel, ConfigDict


class ContentParagraphResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    page: int
    order: int
    kind: str
    text: str
    question_number: int | None = None
    exam_source: str | None = None


class ContentChapterSummary(BaseModel):
    id: str
    number: int
    title: str
    start_page: int
    end_page: int
    paragraph_count: int
    question_count: int


class ContentChapterDetail(ContentChapterSummary):
    paragraphs: list[ContentParagraphResponse]


class ContentBookResponse(BaseModel):
    source_file: str
    book_id: int | None
    title: str
    subtitle: str
    page_count: int
    chapter_count: int
    paragraph_count: int
    question_count: int
    chapters: list[ContentChapterSummary]

File: frontend/src/types/content.ts
export type ContentBlockKind = "paragraph" | "heading" | "question" | "stimulus";

export interface ContentParagraph {
  id: string;
  page: number;
  order: number;
  kind: ContentBlockKind | string;
  text: string;
  question_number: number | null;
  exam_source: string | null;
}

export interface ContentChapterSummary {
  id: string;
  number: number;
  title: string;
  start_page: number;
  end_page: number;
  paragraph_count: number;
  question_count: number;
}

export interface ContentChapterDetail extends ContentChapterSummary {
  paragraphs: ContentParagraph[];
}

export interface ContentBook {
  source_file: string;
  book_id: number | null;
  title: string;
  subtitle: string;
  page_count: number;
  chapter_count: number;
  paragraph_count: number;
  question_count: number;
  chapters: ContentChapterSummary[];
}
